iotdb-core/datanode/src/main/java/org/apache/iotdb/db/queryengine/plan/analyze/AnalyzeUtils.java

```diff
@@ -540,13 +547,14 @@ private static IDPredicate getIdPredicate(
     String rightHandValue;
     if (right instanceof StringLiteral) {
       rightHandValue = ((StringLiteral) right).getValue();
+    } else if (right instanceof NullLiteral) {
+      rightHandValue = null;
```

Comment:
result of XXX = NULL is NULL, so in such case, we should never delete any rows.

Reply:
Done